activitysim/abm/tables/vehicles.py

```diff
@@ -0,0 +1,68 @@
+# ActivitySim
+# See full license in LICENSE.txt.
+import logging
+
+from activitysim.core import inject, pipeline, tracing
+
+logger = logging.getLogger(__name__)
+
+
+@inject.table()
+def vehicles(households):
+    """ Creates the vehicles table and load it as an injectable
+
+    This method initializes the `vehicles` table, where the number of rows
+    is equal to the sum of `households["auto_ownership"]`.
+
+    Parameters
+    ----------
+    households :  orca.DataFrameWrapper
+
+    Returns
+    -------
+    vehicles : pandas.DataFrame
+    """
+
+    # initialize vehicles table
+    vehicles = households.to_frame().loc[
+        households.index.repeat(households['auto_ownership'])]
+    vehicles = vehicles.reset_index()[['household_id']]
+
+    vehicles['vehicle_num'] = vehicles.groupby('household_id').cumcount() + 1
+    # tying the vehicle id to the household id in order to ensure reproducability
+    vehicles['vehicle_id'] = vehicles.household_id * 10 + vehicles.vehicle_num
+    vehicles.set_index('vehicle_id', inplace = True)
+
+    # I do not understand why this line is necessary, it seems circular
```

Comment:
should figure this out before merging

Reply:
Looks like the purpose of this line is to replace the "table" function with the dataframe. Added comment to reflect this just like in persons and households.  This is just a consequence of how the inject table [decorator ](https://github.com/ActivitySim/activitysim/blob/master/activitysim/core/inject.py#L38) and [add_table ](https://github.com/ActivitySim/activitysim/blob/master/activitysim/core/inject.py#L80) functions were constructed.